sink/datadog.go

```diff
@@ -276,9 +289,30 @@ func (s *Datadog) Send(ctx context.Context, m *blip.Metrics) error {
 
 			// Convert Blip metric type to Datadog metric type
 			switch metrics[i].Type {
-			case blip.COUNTER:
+			case blip.CUMULATIVE_COUNTER, blip.DELTA_COUNTER:
+				metricVal := metrics[i].Value
+
+				if s.sendCounterDelta && metrics[i].Type == blip.CUMULATIVE_COUNTER {
+					// if the value is not specifically marked as DELTA, calculate delta
+					metricID := s.uniqIdentifierForMetric(name, tags)
+					val, ok := s.counters[metricID]
+					if !ok {
+						blip.Debug("error calculating delta for the counter: %s, no previous value found, skipping this run (next run will work)", name)
+						s.counters[metricID] = metrics[i].Value
+						continue METRICS
+					}
+					// previous value present, calculate delta
+					delta := metrics[i].Value - val
+					s.counters[metricID] = metrics[i].Value
+					if delta < 0 {
+						blip.Debug("error calculating delta for the counter: %s, got negative delta (can happen due to restart), skipping this run (next run will work).", name)
```

Comment:
If negative due to restart, then probably we return the current value? Granted, that value is partial depending on where/how long into the restarted happened between collection, but partial is still correct and better then null.

Reply:
A loss of metric would aid in detecting an issue though, if it persisted.

Reply:
A loss of metric would probably happen by not receiving the metric at all in the sink which will be detected by no data in DD.

I agree with Dan, we can send the partial value we will get after restart. I will update.